timedctl/helpers.py

```diff
@@ -69,6 +69,21 @@ def time_sum(arr):
     # format as HH:MM:SS
     return str(total)
 
+def activity_sum(arr):
+    """Sum up an array of activities."""
+    total = datetime.timedelta()
+    now = datetime.datetime.now()
+    for line in arr[1:]:
+        start = line[-2]
+        end = line[-1]
+        # if active activity, create end timestamp from `now`
+        if end == "active":
+            total += (now - datetime.datetime.strptime(now.strftime("%F")+"T"+start,"%Y-%m-%dT%H:%M:%S")) 
+        else:
```

Comment:
most likely, by refactoring the calculations above and below this line, you could get rid of the `else` here

because i only gave the code a quick glance, i didn't check the actual formula, but it doesn't look like any overly complicated maths.